lib/sync-DataSetModel.js

```diff
@@ -380,24 +400,53 @@ var self = {
         rec.hash = hash;
         recOut[i] = rec;
       }
-      var globalHash = syncUtil.generateHash(hashes);
 
-      var previousHash = datasetClient.dataHash ? datasetClient.dataHash : '<undefined>';
-      syncUtil.doLog(dataset.id, 'verbose', 'doSyncList cb ' + ( cb !== undefined) + ' - Global Hash (prev :: cur) = ' + previousHash + ' ::  ' + globalHash);
+      console.log('doListHandler recOut - sync : ' + syncUtil.generateHash(recOut));
+      console.log('doListHandler hashes - sync : ' + syncUtil.generateHash(hashes));*/
 
-      datasetClient.dataHash = globalHash;
-      self.datasetClientRecords[datasetClient.id] = recOut;
+      // Start: Aiden Keating - Super amazing work v2
 
-      var res = {
-        hash: globalHash,
-        records: recOut
-      };
+      async.forEachSeries(Object.keys(records), function(recordId, itemCallback){
+        var record = records[recordId];
+        syncUtil.asyncGenerateHash(record, function(err, recordHash) {
+          var rec = {
+            data: record,
+            hash: recordHash
+          };
+          hashes.push(recordHash);
+          recOut[recordId] = rec;
+          itemCallback();
+        });
+      }, function(err) {
+        if (err) return cb(err);
+        // console.log('doListHandler recOut - async : ' + syncUtil.generateHash(recOut));
+        // console.log('doListHandler hashes - async : ' + syncUtil.generateHash(hashes));
+
+        syncUtil.asyncGenerateHash(hashes, function(err, globalHash) {
+          console.log('doListHandler callback - Async : ' + globalHash);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this logging statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS2228)